include/libcyphal/transport/scattered_buffer.hpp

```diff
@@ -0,0 +1,120 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_SCATTERED_BUFFER_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_SCATTERED_BUFFER_HPP_INCLUDED
+
+#include <cetl/any.hpp>
+
+#include <cstdint>
+
+namespace libcyphal
+{
+namespace transport
+{
+
+/// The buffer is movable but not copyable because copying the contents of a buffer is considered wasteful.
+/// The buffer behaves as if it's empty if the underlying implementation is moved away.
+///
+class ScatteredBuffer final
+{
+    // 91C1B109-F90E-45BE-95CF-6ED02AC3FFAA
+    using InterfaceTypeIdType = cetl::
+        type_id_type<0x91, 0xC1, 0xB1, 0x09, 0xF9, 0x0E, 0x45, 0xBE, 0x95, 0xCF, 0x6E, 0xD0, 0x2A, 0xC3, 0xFF, 0xAA>;
+
+public:
+    static constexpr std::size_t ImplementationFootprint = sizeof(void*) * 8;
+
+    class Interface : public cetl::rtti_helper<InterfaceTypeIdType>
+    {
+    public:
+        Interface(const Interface&)            = delete;
+        Interface& operator=(const Interface&) = delete;
+
+        CETL_NODISCARD virtual std::size_t size() const noexcept                      = 0;
+        CETL_NODISCARD virtual std::size_t copy(const std::size_t offset_bytes,
+                                                void* const       destination,
+                                                const std::size_t length_bytes) const = 0;
+
+    protected:
+        Interface()                                = default;
+        Interface(Interface&&) noexcept            = default;
+        Interface& operator=(Interface&&) noexcept = default;
+
```

Comment:
the rule of 5 requires the dtor to be explicitly defaulted

Reply:
will do it at after-next pr